lib/util/pssh.js

```diff
@@ -52,14 +52,14 @@ shaka.util.Pssh = class {
     * */
     this.dataBoundaries = [];
 
-    if (extractMoov) {
+    if (extractFromMoov) {
       new shaka.util.Mp4Parser()
           .box('moov', shaka.util.Mp4Parser.children)
-          .fullBox('pssh', (box) => this.parseBox_(box))
+          .fullBox('pssh', (box) => this.parseBox_(box, /* boxOffset */ 8))
```

Comment:
Instead of passing a hard-coded offset of 8, use `box.start`.  The MP4 parser already supplies this offset for you.

https://shaka-player-demo.appspot.com/docs/api/shaka.extern.html#.ParsedBox

> start | number | The start of this box (before the header) in the original buffer. This start position is the absolute position.

Reply:
I think I am not understanding you well. The parseBox_ method already uses box.start in its implementation.

Reply:
Perhaps I've misunderstood the docs for box.start, but it seems that it supplies you the byte offset of the box.  So why hard-code 8 and add that?

Reply:
I have been investigating this (basically this is taken from the initial PR) and it seems that the problem is related to:

```
 new shaka.util.Mp4Parser()
           .box('moov', shaka.util.Mp4Parser.children)
           .fullBox('pssh', (box) => this.parseBox_(box, /* boxOffset */ 8))
           .parse(psshBox);
```

And the problem, I think is related with the behaviour of `shaka.util.Mp4Parser.children`.

If I understand correctly, that's skipping over the header of the moov atom, which is present if the extractMoov flag is set, but that's not super obvious unless you go up to the constructor of this class.
(extracted from https://github.com/google/shaka-player/pull/2042#discussion_r305484268 )

I'm a bit lost... Any help are welcome. :S

Reply:
@joeyparrish , do you have any idea?

Reply:
I'm not sure what you mean.  I've just gone through the code of `Mp4Parser`, and it seems that when `box.start` always has the offset of that box from the start of the input data buffer.  So you don't need an additional offset AFAICT.  So even if there's a `moov` atom at the front, or the box is the nth child, that should all already be accounted for in `box.start`.

In fact, there's really no need for `extractMoov` as far as I can tell.  It seems that `Mp4Parser` will be happy to parse a buffer starting at `pssh`, even if you first define `moov`.  The box definitions don't dictate the hierarchy.

Reply:
Ok, I found another solution that I think is better and we don't have to have this hardcoded.